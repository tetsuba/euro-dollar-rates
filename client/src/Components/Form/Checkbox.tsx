// TODO: I have added an onChange to the input to stop the react error from firing.
//       Look for a better solution.

import React from 'react'
import {EventClickType} from "../../utils/types";

export interface CheckboxProps {
    key?: string
    label: string,
    checked: boolean,
    onClick: (e: EventClickType) => void,
    index: number,
}

export default function Checkbox({ label, checked, onClick, index }: CheckboxProps): JSX.Element {
    return (
        <div
            className="form-check"
            data-testid="checkbox-custom"
            onClick={onClick}
            data-index={index}
            data-checked={checked}
        >
            <input
                className="form-check-input pe-none"
                type="checkbox"
                checked={checked}
                id="flexCheckDefault"
                onChange={() => {}}
            />
            <label
                className="form-check-label pe-none"
                htmlFor="flexCheckDefault"
            >
                {label}
            </label>
        </div>
    )
}
