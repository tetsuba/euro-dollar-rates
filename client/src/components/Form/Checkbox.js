// TODO: I have added an onChange to the input to stop the react error from firing.
//       Look for a better solution.

import React from 'react'

export default function Checkbox({ label, checked, onClick, index }) {
    return (
        <div
            className="form-check"
            data-testid="checkbox-custom"
            onClick={onClick}
            index={index}
            data-checked={checked}
        >
            <input
                className="form-check-input pe-none"
                type="checkbox"
                checked={checked}
                id="flexCheckDefault"
                onChange={() => {}}
            />
            <label
                className="form-check-label pe-none"
                htmlFor="flexCheckDefault"
            >
                {label}
            </label>
        </div>
    )
}
