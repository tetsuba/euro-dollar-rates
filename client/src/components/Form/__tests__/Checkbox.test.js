import { render, screen, fireEvent } from '@testing-library/react'
import Checkbox from '../Checkbox'

describe('Checkbox', () => {

  const mockProps = {
    key: 'cb-1',
    index: '1',
    label: '2022/06 - $100',
    checked: true,
  }

  describe('render', () => {
    test('checkbox is checked', () => {
      const { asFragment } = render(<Checkbox {...mockProps} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('events', () => {
    test('onclick', () => {
      const mockClickHandler = jest.fn()
      render(<Checkbox onClick={mockClickHandler} />)
      const button = screen.getAllByRole('checkbox')
      fireEvent.click(button[0])
      expect(mockClickHandler).toHaveBeenCalledTimes(1)
    })
  })
})
