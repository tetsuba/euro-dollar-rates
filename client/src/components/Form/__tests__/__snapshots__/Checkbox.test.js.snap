// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox render checkbox is checked 1`] = `
<DocumentFragment>
  <div
    class="form-check"
    data-checked="true"
    data-testid="checkbox-custom"
    index="1"
  >
    <input
      checked=""
      class="form-check-input pe-none"
      id="flexCheckDefault"
      type="checkbox"
    />
    <label
      class="form-check-label pe-none"
      for="flexCheckDefault"
    >
      2022/06 - $100
    </label>
  </div>
</DocumentFragment>
`;
